### PR TITLE
Add GitHub actions workflow files

### PR DESCRIPTION
**Pull request by NarenAnandan**
- Added `main_workflow.yml` to build and publish plugin.
- Added `pr_workflow.yml` to build and test PRs.
- Added `gradle enterprise plugin` to publish build scans to Actions summary.

### CODE REVIEW

File: .github/workflows/main_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Build and Publish
+
+on:
+  push:
+    branches:
+      - 'main'
+
+env:
+  JAVA_VERSION: '18'
```
[the remark this answers is not in the document]
Comment (peter-thomas-mgd) [reply .github/workflows/main_workflow.yml:9]: Ok, I see what happened now. The source compatibility version actually references v1.8, which refers to JRE v8. This means that the generated bytecode will support installing the plugin if it is installed in a legacy environment. Having said that, I don't know whether the version of Gradle and/or Groovy will actually run on Java 8. See the following compatibility matrix for Gradle https://docs.gradle.org/current/userguide/compatibility.html.
TLDR, the build environment should be the same as our standard development environment for all our projects (i.e. Java 17).
